tools/check-enforcer/Azure.Sdk.Tools.CheckEnforcer/GitHubWebhookProcessor.cs

```diff
@@ -0,0 +1,181 @@
+
+            var totalOtherRuns = otherRuns.Count();
+
+            var outstandingOtherRuns = from run in otherRuns
+                                       where run.Conclusion != new StringEnum<CheckConclusion>(CheckConclusion.Success)
+                                       select run;
+
+            var totalOutstandingOtherRuns = outstandingOtherRuns.Count();
+
+            if (totalOtherRuns >= this.ConfigurationStore.MinimumCheckRuns && totalOutstandingOtherRuns == 0)
+            {
+                await client.Check.Run.Update(repositoryId, checkEnforcerRun.Id, new CheckRunUpdate()
+                {
+                    Conclusion = new StringEnum<CheckConclusion>(CheckConclusion.Success)
+                });
+            }
+            else
+            {
+                if (checkEnforcerRun.Conclusion == new StringEnum<CheckConclusion>(CheckConclusion.Success) && totalOutstandingOtherRuns > 0)
+                {
+                    await EnsureCheckEnforcerRunAsync(client, repositoryId, headSha, runs, true);
+                }
+                else
+                {
+                    if (checkEnforcerRun.Status != new StringEnum<CheckStatus>(CheckStatus.InProgress))
+                    {
+                        await client.Check.Run.Update(repositoryId, checkEnforcerRun.Id, new CheckRunUpdate()
+                        {
+                            Status = new StringEnum<CheckStatus>(CheckStatus.InProgress)
+                        });
+                    }
+                }
+
+            }
+        }
+
+        public async Task ProcessWebhookAsync(HttpRequest request, ILogger log, CancellationToken cancellationToken)
+        {
+            if (request.Headers.TryGetValue(GitHubEventHeader, out StringValues eventName))
+            {
+                if (eventName == "check_suite" )
+                {
+                    var payload = await DeserializePayloadAsync<CheckSuiteEventPayload>(request.Body);
+                    var installationId = payload.Installation.Id;
+                    var repositoryId = payload.Repository.Id;
+                    var headSha = payload.CheckSuite.HeadSha;
+
+                    await EvaluateAndUpdateCheckEnforcerRunStatusAsync(installationId, repositoryId, headSha, cancellationToken);
+                }
+                else if (eventName == "check_run")
+                {
+                    var payload = await DeserializePayloadAsync<CheckRunEventPayload>(request.Body);
+                    var installationId = payload.Installation.Id;
+                    var repositoryId = payload.Repository.Id;
+                    var headSha = payload.CheckRun.CheckSuite.HeadSha;
+
+                    await EvaluateAndUpdateCheckEnforcerRunStatusAsync(installationId, repositoryId, headSha, cancellationToken);
+                }
+                else
+                {
+                    throw new GitHubWebhookHandlerNotRegisteredException(eventName);
```

Comment:
Where do you register for the particular event types you want to listen for?

Reply:
This is a bit of a left over from how I was initially planning on writing this. I was thinking of having a handler model where each event had a separate handler. I'm going to rename this exception to be more meaningful.

Reply:
Oh - I might have misunderstood the question. When you register the application you specify the set of events. So I listen to check suite and check run events, as well as PR comments (planning ahead). Once I setup the app in a proper org you'll have access to go in and have a look.